### PR TITLE
 App Inspector UI Changes, Hide Empty Bins, Bin size of 1

### LINKED ISSUE #48: hide columns with 0 items
If I pull something like this, there is a lot of real estate take up by columns that have 0 jobs. Maybe we should tighten this up by removing all columns with 0 jobs.

image

┆Issue is synchronized with this [Asana task](https://app.asana.com/0/1201735133575781/1203168723978571) by [Unito](https://www.unito.io)

### LINKED ISSUE #49: one bin per attribute
When I pull something like this - you can see that I'm at the maximum number of bins. However, I'd like to go even further down to 1 bin per core.  So that every bin is only 1 attribute (core or gpu) wide.

image

┆Issue is synchronized with this [Asana task](https://app.asana.com/0/1201735133575781/1203168725319569) by [Unito](https://www.unito.io)

### PR DESCRIPTION
**Pull request by lukew3**
* Added ability to hide empty bins
* Changed color of app inspector bars
* Rounded app inspector bars and container
* Made x and y labels of app inspector histogram bold
* Moved bins slider to above the histogram
* Added labels and padding to app inspector inputs
* Set bin size to 1 when slider is all the way to the right

Closes #48
Closes #49

┆Issue is synchronized with this [Asana task](https://app.asana.com/0/1135148780858012/1203651069968492) by [Unito](https://www.unito.io)

### CODE REVIEW

File: app/controllers/app_inspector_controller.rb [hunk 1]
```diff
@@ -8,7 +8,11 @@ def index
   def replace_stream
     data = Job.app_inspect(params[:bins_slider].to_i, params[:app_select], params[:cluster_select], params[:property_select])
     opts = {  partial: 'histogram',
-              locals:  { graph_data: data['graph_data'], bin_size: data['bin_size'] } }
+              locals:  {
+                graph_data:      data['graph_data'],
+                bin_size:        data['bin_size'],
+                hide_empty_bins: params[:hide_empty_bins]
```
Comment (johrstrom) [inline app/controllers/app_inspector_controller.rb:14]: I think we should hide empty bins by default. Don't even expose the option.

File: app/assets/stylesheets/app_inspector.scss [hunk 2]
```diff
@@ -10,7 +10,7 @@ p {
 #y_label {
   height: 18pt;
   position: absolute;
-  margin-left: -80px;
+  margin-left: -85px;
   margin-top: 130px;
```
Comment (johrstrom) [inline app/assets/stylesheets/app_inspector.scss:14]: Can we avoid `px` and use `em` or `rem` instead?

File: app/models/job.rb [hunk 3]
```diff
@@ -62,7 +62,13 @@ def app_inspect(bin_count = 5, job_name = 'ondemand/sys/dashboard/sys/bc_osc_jup
         property_hash[job.tres.scan(match_str).flatten.map(&:to_i).sum] += 1
       end
       max_property = property_hash.keys.max + 1
-      bin_size = (max_property.to_f / bin_count).ceil
+      # When slider is all the way to the right, use a bin_size of 1 and set bin_count to max
+      if bin_count == 21
```
Comment (johrstrom) [inline app/models/job.rb:66]: 21 seems to be a special number. Where did we get that number from?

Comment (lukew3) [reply app/models/job.rb:66]: The bin slider is set to choose a number of bins between 1 and 20. I increased the size of the slider to include 21 for the 1 bin/item setting.

Comment (johrstrom) [reply app/models/job.rb:66]: OK I see that now. Hmmmm, OK I thought it was more dynamic than that. I.e., you'd get the options for as many bins as you could.  Cool.

https://github.com/OSC/osc-reporting/blob/d87010d428e1e9e7df04253652e028d539846bf3/app/views/app_inspector/index.html.erb#L26

**Review by johrstrom — APPROVED** [review]

**Comment by johrstrom** [comment]
You can squash all these into 1 commit if you like. If not, I can squash and write the message.

**Pull request merged**